README.md

```diff
@@ -1,44 +1,233 @@
-To view the Analytics template:
+-   [Installing the app using a Scratch Org](#installing-the-app-using-a-scratch-org): This is the recommended installation option. Use this option if you are a developer who wants to experience the app and the code.
 
-'sfdx analytics:template:list'
+-   [Tableau CRM LWC Features](#tableau-crm-lwc-features)
 
-To create an Analytics application from the template you have two options:
+-   [Tableau CRM LWC Metadata](#tableau-crm-lwc-metadata)
 
-From CLI:
+-   [Tableau CRM App Templates](#tableau-crm-app-templates)
 
-'sfdx analytics:app:create -t <templateid>'
+-   [Tableau CRM App Template Commands](#tableau-crm-app-template-commands)
 
-From Analytics Studio
+## Installing the app using a Scratch Org
 
-> Onetime: set environment variable so you always open to Analytics Page
+1. Set up your environment. Follow the steps in the [Quick Start: Lightning Web Components](https://trailhead.salesforce.com/content/learn/projects/quick-start-lightning-web-components/) Trailhead project. The steps include:
 
-export FORCE_OPEN_URL=/analytics/wave/wave.apexp
+    - Enable Dev Hub in your Trailhead Playground
+    - Install Salesforce CLI
+    - Install Visual Studio Code
+    - Install the Visual Studio Code Salesforce extensions, including the Lightning Web Components extension
 
-> Open the scratch org
+1. If you haven't already done so, authorize your hub org and provide it with an alias (**myhuborg** in the command below):
 
-sfdx force:org:open
+    ```
+    sfdx auth:web:login -d -a myhuborg
+    ```
 
-> Select 'Create' > 'App' > 'Start From Template'
+1. Clone the forcedotcom/sfdx-analytics repository:
 
-To to view the Analytics applications from the CLI:
+    ```
+    git clone https://github.com/forcedotcom/sfdx-analytics.git
+    cd sfdx-analytics
+    ```
 
-'sfdx analytics:app:list' 
+1. Create a scratch org and provide it with an alias (sfdx-analytics in the command below):
+
+    ```
+    sfdx force:org:create -s -f config/project-scratch-def.json -a sfdx-analytics
+    ```
+
+1. Push the app to your scratch org:
+
+    ```
+    sfdx force:source:push
+    ```
+
+1. Open the scratch org:
+
+    ```
+    sfdx force:org:open
+    ```
+
+## Tableau CRM LWC Features
+
+**Types**
+
+```
+type Row = {[string]: mixed, ...};
+type Data = Array<Row>;
+type Metadata = {|
+    groups: Array<string>,
+    strings: Array<string>,
+    numbers: Array<string>
+|};
+type Selection = Data;
+type SetSelection = (Selection) => void;
+type SelectMode = 
+    | 'none'
+    | 'single'
+    | 'multi'
+    | 'singlerequired'
+    | 'multirequired';
+
+type State = {|
+    pageId: string,
+    state: DashboardStateJson
+|};
+type GetState = () => State,
+type SetState = (State) => void;
+```
+
+**Data**
+
+Data is the result rows returned by the query as an array of maps.
+```
+[
+    {columnOne: 'one', columnTwo: 123},
+    {columnOne: 'two', columnTwo: 456}
+]
+```
+
+**Metadata**
+
+Metadata describes the shape of the results.
```

Comment:
We should include "groups" here as well. This is quite important for implementing selection logic like in the list example.